### DISCUSSION TITLE
v0.2.4 with out 3d pack.

### DISCUSSION THREAD

**Discussion by Falkonar** [YanWenKun/ComfyUI-Windows-Portable#28] [Q&A] [answered]
I tried just running this version but it freeze as soon as I move the nodes . I mean it starts, but I can't work in it, the browser freezes. Windows 10.

**Comment by YanWenKun** [comment]
Try the new UI: https://blog.comfy.org/comfyui-v1-release/#enabling-new-ui

And, what is the `v0.2.4` mentioning about?

**Comment by YanWenKun** [comment]
> my main task is to run SDXL_EcomID_ComfyUI, which I can't run on the official build without torch compilation.

It's `insightface` which needs compile-install. 
If you're using [2024/10/25 Without 3D-Pack](https://github.com/YanWenKun/ComfyUI-Windows-Portable/releases/tag/v5.2-no3d) or [ComfyUI official portable pack](https://github.com/comfyanonymous/ComfyUI/releases/tag/v0.2.4), you can just install a compiled one:

```
python.exe -s -m pip install https://raw.githubusercontent.com/Gourieff/Assets/refs/heads/main/Insightface/insightface-0.7.3-cp312-cp312-win_amd64.whl
```

If you want a bundled one, use [2024/10/18](https://github.com/YanWenKun/ComfyUI-Windows-Portable/releases/tag/v5.1), it has `insightface` installed.

**Comment by YanWenKun** [comment]
And for new UI, the ComfyUI V1 is officially released. You can enable it in the [settings](https://blog.comfy.org/comfyui-v1-release/#enabling-new-ui). I guess this is not related to your issue, but it's worth trying if the browser freezes again.

**Comment by Falkonar** [comment]
![image](https://github.com/user-attachments/assets/c961cdcc-3b58-47f0-bf8b-dae179d9dcaf)
Thanks for the advice, I will try to install insightface . But the interface still freeze and I can't move the nodes. Only to press the menu buttons

**Comment by YanWenKun** [comment]
![sc](https://github.com/user-attachments/assets/d33b4f9e-9ffb-4b1f-b3d6-426762a3e6ad)


OK this has nothing to do with the new UI. Two parts of setup:

#### 1.

To solve the browser freezing, we need to simplify the `custom_nodes`, keep only what we need.

In order to run the example workflow, you only need:

- SDXL_EcomID_ComfyUI
- PuLID_ComfyUI
- ComfyUI_InstantID
- And optionally ComfyUI-Manager

You can just rename the folder to `custom_nodes_bak` or whatever you like. Make a new one and only keep the items mentioned above.

Otherwise try using your browser's "private mode". Or try other browsers like Firefox, 

#### 2.

To solve the dependency of SDXL_EcomID_ComfyUI, some packages needs to be installed manually. The full steps:

```cmd
python.exe -s -m pip install https://raw.githubusercontent.com/Gourieff/Assets/refs/heads/main/Insightface/insightface-0.7.3-cp312-cp312-win_amd64.whl

python.exe -s -m pip install git+https://github.com/rodjjo/filterpy.git

python.exe -s -m pip install facexlib
```

Then you should have it running.

(Ref on filterpy: https://stackoverflow.com/questions/78216797/failed-to-install-filterpy )

**Comment by Falkonar** [comment]
https://github.com/YanWenKun/ComfyUI-Windows-Portable/releases/tag/v5.1 Did you test it on this build?

**Comment by YanWenKun** [comment]
> https://github.com/YanWenKun/ComfyUI-Windows-Portable/releases/tag/v5.1 Did you test it on this build?

No I test it on https://github.com/YanWenKun/ComfyUI-Windows-Portable/releases/tag/v5.2-no3d

Should work the same?

**Comment by Falkonar** [comment]
> > https://github.com/YanWenKun/ComfyUI-Windows-Portable/releases/tag/v5.1 Did you test it on this build?
> 
> No I test it on https://github.com/YanWenKun/ComfyUI-Windows-Portable/releases/tag/v5.2-no3d
> 
> Should work the same?

I have a feeling something is wrong with windows 10. Strangely everything was fine before. Version v5.2-no3d never managed to run without errors. torch not compiled. I just downloaded v5.1 and I just can't get it to run. 

![image](https://github.com/user-attachments/assets/865c583f-37b1-434a-a4be-f3ac6f145b7c)

**Comment by Falkonar** [comment]
![image](https://github.com/user-attachments/assets/72011719-c873-450f-b22d-93993b147089)
This is my attempt with v5.2-no3d

**Comment by YanWenKun** [comment]
The v5.1 pic looks weird, the log only shows warning message (about deprecation), no errors, but it stopped running?

The v5.2-no3d pic shows a possible dep issue:

```
python.exe -s -m pip install --force-reinstall onnx
```

**Comment by Falkonar** [comment]
> The v5.1 pic looks weird, the log only shows warning message (about deprecation), no errors, but it stopped running?
> 
> The v5.2-no3d pic shows a possible dep issue:
> 
> ```
> python.exe -s -m pip install --force-reinstall onnx
> ```

![image](https://github.com/user-attachments/assets/7624a767-e529-40af-a09a-28a51db16b63)
v5.2-no3d

Yes The v5.1 just first start ends with no error and exit.

**Comment by Falkonar** [comment]
I will wait for you update. It request to recompile my torch.... ;(

**Comment by YanWenKun** [comment]
I'm compiling a new all-in-one package for release...

I guess [CUDA Toolkit](https://developer.nvidia.com/cuda-downloads?target_os=Windows&target_arch=x86_64&target_version=10&target_type=exe_local) and [VS 2022 with C++](https://visualstudio.microsoft.com/downloads/?q=build+tools) is only used by Comfy3D in the context of ComfyUI. Maybe some new nodes are using them as well.

Another approach is to use [Conda](https://docs.comfy.org/get_started/manual_install) that you build the environment step by step. Conda could avoid some issues like "No module named ABCDEFG".

**Comment by Falkonar** [comment]
I look forward to it! 
![image](https://github.com/user-attachments/assets/870e65a5-2301-4d67-8efd-a2694cfb3783)
Did you use Conda for your install? I just tried building on the original ComfyUI newest build, but I'm running into Torch not compiled with CUDA enabled. Can you tell me which version torchvision xformers & torch? Maybe I can compile it myself?

**Comment by YanWenKun** [comment]
This "Torch not compiled with CUDA enabled" message is indicating you have PyTorch CPU installed. Check the PyTorch official page to install PyTorch CUDA: https://pytorch.org/get-started/locally/

```
conda install pytorch torchvision torchaudio pytorch-cuda=12.1 -c pytorch -c nvidia
```

```
conda install pytorch torchvision torchaudio pytorch-cuda=12.4 -c pytorch -c nvidia
```

**Comment by YanWenKun** [comment]
Check out the new release:
https://github.com/YanWenKun/ComfyUI-Windows-Portable/releases/tag/v6.0

**Comment by Falkonar** [comment]
> Check out the new release: https://github.com/YanWenKun/ComfyUI-Windows-Portable/releases/tag/v6.0
maybe there's a problem with the paths?
![image](https://github.com/user-attachments/assets/73753561-01cd-4626-9d77-f7f99ed24c78)

LLM tell me to find - "G:\ComfyUI_Windows_portable\python_embeded\Lib\site-packages\kornia\feature\lightglue.py" and change - @torch.cuda.amp.custom_fwd(cast_inputs=torch.float32) to @torch.amp.custom_fwd(cast_inputs=torch.float32, device_type='cuda') . But this just change situation to this - 
![image](https://github.com/user-attachments/assets/ed06ee13-399a-47d7-a063-bff46ebe6064)

**Comment by YanWenKun** [comment, marked as answer]
This would need verbose logging and Python debugging. But I recommend just try creating a clean environment at first. The shutdown was too weired...

**Reply by Falkonar** [reply]
I have install fresh windows and your build run excellent. The problem was with my win environment echo%path% was a mess

**Reply by YanWenKun** [reply]
Nice job getting everything sorted!

**Comment by Falkonar** [comment]
> This would need verbose logging and Python debugging. But I recommend just try creating a clean environment at first. The shutdown was too weired...

To my mind comfy embedded conflicts with windows py environment. Why I can not run your build isolated? In portable mode. I will install windows 10 clean install and try. Cuda 12.4.1 & Ms build 2002 desktop+ tools. And do I need to run ComfyUI portable from conda inviroment? I thought that I could use your python embedded inside comfy and use - python.exe -m -s install -r (that way)

Second question, maybe I could use docker env in that situation?

**Comment by YanWenKun** [comment]
1. You don't need Conda for that, just PIP is fine.

2. Yes, I definitely recommend using Docker or Podman, whether on WSL2 or Linux. It may be more complicated, but it provides better isolation.

https://github.com/YanWenKun/ComfyUI-Docker

**Comment by Falkonar** [comment]
xformers 0.0.28.post2 requires torch==2.5.0, but you have torch 2.5.1+cu124 which is incompatible

what version of torch should I choose with cu124 to work with 28.post2 ?

On the PyTorch web I can download 2.5.1 or 2.4.1 . I can not find 2.5.0 that request for xformers

**Comment by YanWenKun** [comment]
To use 3D-Pack, you have to use specific version of xFormers + PyTorch.

If not using nodes that require xFormers, just install latest PyTorch.

**Comment by YanWenKun** [comment]
This topic is more than a issue. I'm converting it to a Discussion topic and continue discussion 😆
